pegasus/router.rb

```diff
@@ -171,6 +171,7 @@ def self.set_max_age(type, default)
   get '*' do |uri|
     pass unless path = resolve_static('public', uri)
     cache :static
+    NewRelic::Agent.set_transaction_name(uri) if defined? NewRelic
```

Comment:
This particular route only resolves to static files located on the filesystem, so it won't ever resolve to any unique ID in the path.

The other route potentially resolves dynamic paths to files located on the filesystem with the `splat` suffix, through some special logic. I've added a commit to handle this special case (and added a test to ensure correctness).